src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -118,8 +122,14 @@ public static partial class ImmutableArray
         public System.Collections.Immutable.ImmutableArray<T> Add(T item) { throw null; }
         public System.Collections.Immutable.ImmutableArray<T> AddRange(System.Collections.Generic.IEnumerable<T> items) { throw null; }
         public System.Collections.Immutable.ImmutableArray<T> AddRange(System.Collections.Immutable.ImmutableArray<T> items) { throw null; }
+        public System.Collections.Immutable.ImmutableArray<T> AddRange(System.ReadOnlySpan<T> items) { throw null; }
+        public System.Collections.Immutable.ImmutableArray<T> AddRange(params T[] items) { throw null; }
         public System.ReadOnlyMemory<T> AsMemory() { throw null; }
         public System.ReadOnlySpan<T> AsSpan() { throw null; }
+        public System.ReadOnlySpan<T> AsSpan(int start, int length) { throw null; }
+#if NETCOREAPP3_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
```

Comment:
We generally avoid using ifdefs in ref source files. Instead we use partial classes with tfm-specific APIs residing in separate files. See 

https://github.com/dotnet/runtime/tree/main/src/libraries/System.Net.Http.Json/ref

for an example of the approach.

Reply:
Separated, thanks.